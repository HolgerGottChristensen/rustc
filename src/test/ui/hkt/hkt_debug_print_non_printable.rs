// HKT test that something cloneable can be called with clone, method call syntax, definition and call.
// check-fail

use std::fmt::{Debug};

fn test<I<%J>: Debug>(input: I<u32>) {
    println!("Hej verden: {:?}", input);
}

fn main() {
    test::<Option<%J>>(Some(42)); //~ ERROR `%J` doesn't implement `Debug`
}
