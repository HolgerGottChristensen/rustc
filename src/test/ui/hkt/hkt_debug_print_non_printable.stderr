error[E0277]: `%J` doesn't implement `Debug`
  --> $DIR/hkt_debug_print_non_printable.rs:11:12
   |
LL |     test::<Option<%J>>(Some(42));
   |            ^^^^^^^^^^ `%J` cannot be formatted using `{:?}` because it doesn't implement `Debug`
   |
   = help: the trait `Debug` is not implemented for `%J`
   = help: the trait `Debug` is implemented for `Option<T>`
   = note: required for `Option<%J>` to implement `Debug`
note: required by a bound in `test`
  --> $DIR/hkt_debug_print_non_printable.rs:6:16
   |
LL | fn test<I<%J>: Debug>(input: I<u32>) {
   |                ^^^^^ required by this bound in `test`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
